cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLGenerator.scala

```diff
@@ -0,0 +1,654 @@
+    }
+    sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(wordsArr.length - 1)}'")
+    sb.toString()
+  }
+
+  protected def parseNumberRelation(filter: FilterStatement,
+                                    fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.inRange =>
+        if (filter.values.size != 2) throw new QueryParsingException(s"relation: ${filter.relation} require two parameters")
+        s"$fieldExpr >= ${filter.values(0)} and $fieldExpr < ${filter.values(1)}"
+      case Relation.in =>
+        s"$fieldExpr in ( ${filter.values.mkString(",")} )"
+      case _ =>
+        s"$fieldExpr ${filter.relation} ${filter.values(0)}"
+    }
+  }
+
+  protected def parseJsonRelation(filter: FilterStatement, fieldExpr: String): String = {
+    filter.relation match  {
+      case Relation.inRange =>
+        if (filter.values.size != 2) throw new QueryParsingException(s"relation: ${filter.relation} require two parameters")
+        s"$fieldExpr >= ${filter.values(0)} and $fieldExpr < ${filter.values(1)}"
+      case Relation.in =>
+        filter.values(0) match {
+          case i: Integer =>
+            s"""$fieldExpr in ( ${filter.values.mkString(",")} )"""
+          case s: String =>
+            s"""$fieldExpr in ( "${filter.values.mkString("\",\"")}" )"""
+        }
+      case Relation.contains =>
+        val sb = new StringBuilder
+        val wordsArr = ArrayBuffer[String]()
+        filter.values.foreach(w => wordsArr += "%" + w + "%")
+        for (i <- 0 until (wordsArr.length - 1)){
+          sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(i)}' and ")
+        }
+        sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(wordsArr.length - 1)}'")
+        sb.toString()
+      case Relation.matches => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr="${values(0)}""""
+      }
+      case _ => {
+        s"$fieldExpr ${filter.relation} ${filter.values(0)}"
+      }
+    }
+  }
+
+  protected def parseAggregateFunc(aggr: AggregateStatement,
+                                   fieldExpr: String): String = {
+    def aggFuncExpr(aggFunc: String): String = {
+      if (aggr.field.name.equals("*")) {
+        s"$aggFunc($groupVar)"
+      } else {
+        s"$aggFunc($fieldExpr)"
+      }
+    }
+
+    aggr.func match {
+      case topK: TopK => ???
+      case DistinctCount => ???
+      case _ =>
+        aggFuncExpr(getAggregateStr(aggr.func))
+    }
+  }
+
+  private def parseFilter(filters: Seq[FilterStatement], exprMap: Map[String, FieldExpr], unnestTestStrs: Seq[String], queryBuilder: StringBuilder): ParsedResult = {
+    if (filters.isEmpty && unnestTestStrs.isEmpty) {
+      ParsedResult(Seq.empty, exprMap)
+    } else {
+      val filterStrs = filters.map { filter =>
+        filter.field.dataType match {
+          case js: DataType.Json.type if (filter.field.name.contains(".")) =>
```

Comment:
Calling subfield of Json, I'm considering adding a function only for Json's field's transform, so this line can stop repeating.